src/main.rs

```diff
@@ -177,25 +186,11 @@ async fn collect_from_pods(
         .list(&ListParams::default())
         .await
         .with_context(|| "Failed to list pods via k8s api".to_string())?;
-    for pod in pods.items.into_iter().filter(|pod| {
-        pod.status
-            .as_ref()
-            .and_then(|ps| ps.phase.as_ref().map(|s| s == "Running"))
-            .unwrap_or(false)
-    }) {
-        let status = pod.status.as_ref();
+    for pod in pods.items.into_iter().filter(is_scheduled) {
         let spec = pod.spec.as_ref();
-        let node_name = status
-            .and_then(|v| v.nominated_node_name.clone())
-            .or_else(|| spec.and_then(|m| m.node_name.clone()));
+        let node_name = spec.and_then(|m| m.node_name.clone());
         let containers = spec.map(|s| s.containers.clone()).unwrap_or_else(|| vec![]);
-        for (_, container) in containers.into_iter().enumerate().filter(|(i, _)| {
-            status
-                .and_then(|s| s.container_statuses.as_ref())
-                .and_then(|v| v.get(*i).and_then(|v| v.state.as_ref()))
-                .map(|s| s.running.is_some())
-                .unwrap_or(true)
-        }) {
```

Comment:
This filter is to avoid to count resources (CPU, memory,...) used by non-running container like thoses used for init phase.

Reply:
The "containers" field that is being filtered is only the regular containers.  It does not include the initContainers.  Also the initContainer status information is not there.  It is in initContainerStatuses.  See the associated issue for an example.

Reply:
I think this is resolved based on your latest comment but correct me if I am wrong.